From 96bb1e2dec09922985b6cd4813600d13f72d475a Mon Sep 17 00:00:00 2001
From: Tanghongwei <tanghongwei2024@outlook.com>
Date: Sat, 20 Sep 2025 14:29:51 +0800
Subject: [PATCH 7/8] repet

---
 perip/uart16550/rtl/uart_tfifo.v | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/perip/uart16550/rtl/uart_tfifo.v b/perip/uart16550/rtl/uart_tfifo.v
index bc715ff7..7eed245a 100644
--- a/perip/uart16550/rtl/uart_tfifo.v
+++ b/perip/uart16550/rtl/uart_tfifo.v
@@ -211,8 +211,8 @@ begin
             begin
                 top   <= #1 top_plus_1;
                 count <= #1 count + 1'b1;
-                $write("%c", data_in);
-                $fflush();
+                // $write("%c", data_in);
+                // $fflush();
             end
         2'b01 : if(count>0)
             begin
@@ -222,8 +222,8 @@ begin
         2'b11 : begin
                 bottom <= #1 bottom + 1'b1;
                 top    <= #1 top_plus_1;
-                $write("%c", data_in);
-                $fflush();
+                // $write("%c", data_in);
+                // $fflush();
                 end
         default: ;
         endcase
-- 
2.34.1

