From 0e232dfa1564078e89d1cc60603dd574ec9f3b48 Mon Sep 17 00:00:00 2001
From: Zihao Yu <yuzihao@ict.ac.cn>
Date: Fri, 27 Aug 2021 21:49:16 +0800
Subject: [PATCH 08/74] chiplink: use smaller SRAM

---
 src/chiplink/Parameters.scala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/chiplink/Parameters.scala b/src/chiplink/Parameters.scala
index cc677c29..cdc1043e 100644
--- a/src/chiplink/Parameters.scala
+++ b/src/chiplink/Parameters.scala
@@ -22,8 +22,8 @@ case class ChipLinkParams(TLUH: Seq[AddressSet], TLC: Seq[AddressSet], sourceBit
   val clSourceBits = 16
   val clSinkBits = 16
   val crossing = AsyncQueueParams()
-  val Qdepth = 8192 / dataBytes
-  val maxXfer = 4096
+  val Qdepth = 128 / dataBytes
+  val maxXfer = 64
   val xferBits = log2Ceil(maxXfer)
   val creditBits = 20 // use saturating addition => we can exploit at most 1MB of buffers
   val addressBits = 64
-- 
2.34.1

