From 3d653737e88672d02a94d0136a60488de4fd386c Mon Sep 17 00:00:00 2001
From: Haojin Tang <tanghaojin@outlook.com>
Date: Sat, 25 Sep 2021 22:39:05 +0800
Subject: [PATCH 29/74] perip,spi: support fast simulation

---
 perip/spi/rtl/spi_top_apb.v | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/perip/spi/rtl/spi_top_apb.v b/perip/spi/rtl/spi_top_apb.v
index a04874b6..883c54e3 100644
--- a/perip/spi/rtl/spi_top_apb.v
+++ b/perip/spi/rtl/spi_top_apb.v
@@ -1,3 +1,7 @@
+// define this macro to enable fast behavior simulation
+// for flash by skipping SPI transfers
+//`define FAST_FLASH
+
 module spi_top_apb #(
   parameter flash_addr_start = 32'h30000000,
   parameter flash_addr_end   = 32'h3fffffff,
@@ -23,6 +27,26 @@ module spi_top_apb #(
   output                  spi_irq_out
 );
 
+`ifdef FAST_FLASH
+
+wire [31:0] data;
+parameter invalid_cmd = 8'h0;
+flash_cmd flash_cmd_i(
+  .clock(clk),
+  .valid(in_psel && !in_penable),
+  .cmd(in_pwrite ? invalid_cmd : 8'h03),
+  .addr({8'b0, in_paddr[23:2], 2'b0}),
+  .data(data)
+);
+assign spi_sck    = 1'b0;
+assign spi_ss     = 2'b0;
+assign spi_mosi   = 1'b1;
+assign in_pslverr = 1'b0;
+assign in_pready  = in_penable && in_psel && !in_pwrite;
+assign in_prdata  = data[31:0];
+
+`else
+
 wire [7:0] ss_pad_o;
 assign spi_ss = ss_pad_o[spi_ss_num-1:0];
 
@@ -46,4 +70,6 @@ spi_top u0_spi_top (
   .miso_pad_i(spi_miso)
 );
 
+`endif // FAST_FLASH
+
 endmodule
-- 
2.34.1

