From 965e78a168f9f148331c4dbce4f72dc4bf200a43 Mon Sep 17 00:00:00 2001
From: Zihao Yu <yuzihao@ict.ac.cn>
Date: Mon, 4 Dec 2023 10:04:45 +0800
Subject: [PATCH 32/74] SoC: add 8KB AXI4SRAM

---
 src/SoC.scala | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/SoC.scala b/src/SoC.scala
index 05063e16..5b6f23c2 100644
--- a/src/SoC.scala
+++ b/src/SoC.scala
@@ -36,9 +36,10 @@ class ysyxSoCASIC(implicit p: Parameters) extends LazyModule {
     AddressSet.misaligned(0x30000000, 0x10000000)   // XIP flash
   ))
   val lmrom = LazyModule(new AXI4MROM(AddressSet.misaligned(0x20000000, 0x1000)))
+  val sramNode = AXI4RAM(AddressSet.misaligned(0x0f000000, 0x2000).head, false, true, 8, None, Nil, false)
 
   List(lspi.node, luart.node).map(_ := apbxbar)
-  List(chiplinkNode, apbxbar := AXI4ToAPB(), lmrom.node).map(_ := xbar)
+  List(chiplinkNode, apbxbar := AXI4ToAPB(), lmrom.node, sramNode).map(_ := xbar)
   xbar := cpu.masterNode
 
   override lazy val module = new Impl
-- 
2.34.1

