From 3f411d5a41841f37aeefb963cd44976a99449e78 Mon Sep 17 00:00:00 2001
From: Zihao Yu <yuzihao@ict.ac.cn>
Date: Mon, 11 Dec 2023 12:55:40 +0800
Subject: [PATCH 34/74] device,MROM: add assertion for write operations

---
 src/device/MROM.scala | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/device/MROM.scala b/src/device/MROM.scala
index 442e86e1..134ecfc2 100644
--- a/src/device/MROM.scala
+++ b/src/device/MROM.scala
@@ -67,5 +67,8 @@ class AXI4MROM(address: Seq[AddressSet])(implicit p: Parameters) extends LazyMod
     in.aw.ready := false.B
     in. w.ready := false.B
     in. b.valid := false.B
+
+    assert(!in.aw.valid, "do not support write operations")
+    assert(!in. w.valid, "do not support write operations")
   }
 }
-- 
2.34.1

