From 51500b51a631d04e4eade7140fe7b124a2c14a60 Mon Sep 17 00:00:00 2001
From: Zihao Yu <yuzihao@ict.ac.cn>
Date: Mon, 11 Dec 2023 12:56:09 +0800
Subject: [PATCH 35/74] device,MROM: put comment into the message of assertion

---
 src/device/MROM.scala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/device/MROM.scala b/src/device/MROM.scala
index 134ecfc2..da0a59a1 100644
--- a/src/device/MROM.scala
+++ b/src/device/MROM.scala
@@ -56,7 +56,7 @@ class AXI4MROM(address: Seq[AddressSet])(implicit p: Parameters) extends LazyMod
     mrom.io.raddr := in.ar.bits.addr
     mrom.io.ren := in.ar.fire
     in.ar.ready := true.B
-    assert(!(in.ar.fire && in.ar.bits.size === 3.U)) // do not support 8 byte transfter
+    assert(!(in.ar.fire && in.ar.bits.size === 3.U), "do not support 8 byte transfter")
 
     in.r.bits.data := RegEnable(Fill(2, mrom.io.rdata), in.ar.fire)
     in.r.bits.id := RegEnable(in.ar.bits.id, in.ar.fire)
-- 
2.34.1

