From dc9634b3ddb9c3869f15289f92f2fe37a0a14682 Mon Sep 17 00:00:00 2001
From: Zihao Yu <yuzihao@ict.ac.cn>
Date: Wed, 10 Jan 2024 18:18:10 +0800
Subject: [PATCH 58/74] SoC: expose uart port

---
 src/SoC.scala | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/SoC.scala b/src/SoC.scala
index 3d339ce6..086c8e45 100644
--- a/src/SoC.scala
+++ b/src/SoC.scala
@@ -121,7 +121,6 @@ class ysyxSoCFull(implicit p: Parameters) extends LazyModule {
     }
 
     masic.intr_from_chipSlave := false.B
-    masic.uart.rx := false.B
 
     val flash = Module(new flash)
     flash.io <> masic.spi
@@ -140,9 +139,11 @@ class ysyxSoCFull(implicit p: Parameters) extends LazyModule {
       val gpio = chiselTypeOf(masic.gpio)
       val ps2 = chiselTypeOf(masic.ps2)
       val vga = chiselTypeOf(masic.vga)
+      val uart = chiselTypeOf(masic.uart)
     })
     externalPins.gpio <> masic.gpio
     externalPins.ps2 <> masic.ps2
     externalPins.vga <> masic.vga
+    externalPins.uart <> masic.uart
   }
 }
-- 
2.34.1

