From 287645fcaaf344306591dbd6941f82d43a5f1920 Mon Sep 17 00:00:00 2001
From: Zihao Yu <yuzihao@ict.ac.cn>
Date: Fri, 2 Feb 2024 22:14:24 +0800
Subject: [PATCH 60/74] device,MROM: do not allow outstanding requests

---
 src/device/MROM.scala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/device/MROM.scala b/src/device/MROM.scala
index da0a59a1..a5bd1374 100644
--- a/src/device/MROM.scala
+++ b/src/device/MROM.scala
@@ -55,7 +55,7 @@ class AXI4MROM(address: Seq[AddressSet])(implicit p: Parameters) extends LazyMod
 
     mrom.io.raddr := in.ar.bits.addr
     mrom.io.ren := in.ar.fire
-    in.ar.ready := true.B
+    in.ar.ready := (state === stateIdle)
     assert(!(in.ar.fire && in.ar.bits.size === 3.U), "do not support 8 byte transfter")
 
     in.r.bits.data := RegEnable(Fill(2, mrom.io.rdata), in.ar.fire)
-- 
2.34.1

