From aa30ed0860987168a517f92dc9fa791f63a5d041 Mon Sep 17 00:00:00 2001
From: Zihao Yu <yuzihao@ict.ac.cn>
Date: Sat, 3 May 2025 11:54:22 +0800
Subject: [PATCH 70/74] SoC: add AXI4Buffer to easily cut combinational loop

---
 src/SoC.scala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/SoC.scala b/src/SoC.scala
index 4be26359..1c65cd87 100644
--- a/src/SoC.scala
+++ b/src/SoC.scala
@@ -48,7 +48,7 @@ class ysyxSoCASIC(implicit p: Parameters) extends LazyModule {
   val lsdram_axi = if ( Config.sdramUseAXI) Some(LazyModule(new AXI4SDRAM(sdramAddressSet))) else None
 
   List(lspi.node, luart.node, lpsram.node, lgpio.node, lkeyboard.node, lvga.node).map(_ := apbxbar)
-  List(apbxbar := APBDelayer() := AXI4ToAPB(), lmrom.node, sramNode).map(_ := xbar2)
+  List(apbxbar := APBDelayer() := AXI4ToAPB() := AXI4Buffer(), lmrom.node, sramNode).map(_ := xbar2)
   xbar2 := AXI4UserYanker(Some(1)) := AXI4Fragmenter() := xbar
   if (Config.sdramUseAXI) lsdram_axi.get.node := ysyx.AXI4Delayer() := xbar
   else                    lsdram_apb.get.node := apbxbar
-- 
2.34.1

